### DISCUSSION TITLE
How can I get real port number?

### DISCUSSION THREAD

**Discussion by jsjclink** [ANLAB-KAIST/KENSv3#69] [Project Q&A 2. Transport Layer (TCP)] [answered]
In https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Figuring-out-the-source-IP-address-to-use
I could figure out the src_ip. But, I think the port number which obtained from (int port = getRoutingTable(ip);) is not real port number. (the result is 0)
Is it the real port number?
If not, how can I get real port number? Is it just ok to use any port number in this assignment?

**Comment by Pusnow** [comment, marked as answer]
In network area, there are two *port*s.

1. Port in TCP: pointing application in a host
2. Port in NIC: hardware interface

The `getRoutingTable()` returns the second version of port number. 

I guess you mean the *real port number* is the first one.
If you need new port number (the first one), you can randomly allocate new one.

**Reply by hy2850** [reply]
If client host calls `connect()` (CONNECT syscall), how can it figure out its IP (source IP)? IP/Port for the destination socket is given in the 'addr' argument, but the source IP is unknown. Are we supposed to use the `NIC port` in this case?

**Comment by starrybamboo** [comment]
> If client host calls `connect()` (CONNECT syscall), how can it figure out its IP (source IP)? IP/Port for the destination socket is given in the 'addr' argument, but the source IP is unknown. Are we supposed to use the `NIC port` in this case?

I also have a question about how to find the source IP. Have you solved it?
